silkworm/silkrpc/core/logs_walker.cpp

```diff
@@ -0,0 +1,217 @@
+        auto block_hash_bytes = silkworm::from_hex(filter.block_hash.value());
+        if (!block_hash_bytes.has_value()) {
+            start = end = std::numeric_limits<uint64_t>::max();
+        } else {
+            auto block_hash = silkworm::to_bytes32(block_hash_bytes.value());
+            auto block_number = co_await core::rawdb::read_header_number(tx_database_, block_hash);
+            start = end = block_number;
+        }
+    } else {
+        uint64_t last_executed_block_number = std::numeric_limits<uint64_t>::max();
+        if (filter.from_block.has_value()) {
+            start = co_await core::get_block_number(filter.from_block.value(), tx_database_);
+        } else {
+            last_executed_block_number = co_await core::get_latest_executed_block_number(tx_database_);
+            start = last_executed_block_number;
+        }
+        if (filter.to_block.has_value()) {
+            end = co_await core::get_block_number(filter.to_block.value(), tx_database_);
+        } else {
+            if (last_executed_block_number == std::numeric_limits<uint64_t>::max()) {
+                last_executed_block_number = co_await core::get_latest_executed_block_number(tx_database_);
+            }
+            end = last_executed_block_number;
+        }
+    }
+    co_return std::make_pair(start, end);
+}
+
+Task<void> LogsWalker::get_logs(std::uint64_t start, std::uint64_t end,
+                                const FilterAddresses& addresses, const FilterTopics& topics, const LogFilterOptions& options, bool desc_order, std::vector<Log>& logs) {
+    SILK_INFO << "start block: " << start << " end block: " << end;
+
+    const auto chain_storage{tx_database_.get_tx().create_storage(tx_database_, backend_)};
+    roaring::Roaring block_numbers;
+    block_numbers.addRange(start, end + 1);  // [min, max)
+
+    SILK_DEBUG << "block_numbers.cardinality(): " << block_numbers.cardinality();
+
+    if (!topics.empty()) {
+        auto topics_bitmap = co_await ethdb::bitmap::from_topics(tx_database_, db::table::kLogTopicIndexName, topics, start, end);
+        SILK_TRACE << "topics_bitmap: " << topics_bitmap.toString();
+        if (topics_bitmap.isEmpty()) {
+            block_numbers = topics_bitmap;
+        } else {
+            block_numbers &= topics_bitmap;
+        }
+    }
+    SILK_DEBUG << "block_numbers.cardinality(): " << block_numbers.cardinality();
+    SILK_TRACE << "block_numbers: " << block_numbers.toString();
+
+    if (!addresses.empty()) {
+        auto addresses_bitmap = co_await ethdb::bitmap::from_addresses(tx_database_, db::table::kLogAddressIndexName, addresses, start, end);
+        if (addresses_bitmap.isEmpty()) {
+            block_numbers = addresses_bitmap;
+        } else {
+            block_numbers &= addresses_bitmap;
+        }
+    }
+    SILK_DEBUG << "block_numbers.cardinality(): " << block_numbers.cardinality();
+    SILK_TRACE << "block_numbers: " << block_numbers.toString();
```

Comment:
Unify these two log traces into one at TRACE level

Reply:
I think this is correct as it could be useful to see at DEBUG level just the cardinality and at TRACE one all the block numbers (and the old code was in this way...)